### PR TITLE
sorter/leveldb: add output buffer

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add outputBuffer a struct that facilitate leveldb table sorter https://github.com/pingcap/ticdc/pull/3492.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by zhaoxinyu — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

File: cdc/sorter/leveldb/buffer.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		if b.resolvedEvents[0].CRTs != event.CRTs {
+			log.Panic("commit ts must be equal",
+				zap.Uint64("newCommitTs", event.CRTs),
+				zap.Uint64("commitTs", b.resolvedEvents[0].CRTs))
+		}
+	}
+	b.resolvedEvents = append(b.resolvedEvents, event)
+}
+
+// appendResolvedEvent appends resolved events to the buffer.
```
Comment (asddongmen) [inline cdc/sorter/leveldb/buffer.go:99]: typos？

**Review by asddongmen — APPROVED** [review]

**Pull request merged**